### ISSUE TITLE
Bad "Horizontal/Vertical stack cards" after 2024.3.0 update

### ISSUE THREAD

**Issue by pickonedev** [home-assistant/frontend#20192]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

On some devices/computers, everything works well, but on my older huawei tablets with android 10, the vertical and horizontal stack are doing something strange, are cutting the padding around elements, or something similar.

This is how the page should look and it is looking on some newer tablets (Android 12)
![316887081-51d8bbf1-3513-4800-8832-4ba41b39c14b](https://github.com/home-assistant/frontend/assets/34201387/efc2babd-72ac-4800-b760-c1774061dad1)

And this is how the vertical and even horizontal stack, it is acting on older Android 10 tablets
![316887124-56686225-5dec-43f8-a19e-c6b56bd3470d](https://github.com/home-assistant/frontend/assets/34201387/25cfe89f-bc13-4f75-b890-c9306126c6c9)

I waited new updates but nothing repaired the problem, I deleted browser cache and all, still nothing.
I am using Fully Kiosk Browser on my all devices.

Any solution?

### Describe the behavior you expected

It should look exactly like in the first image, on all the devices

### Steps to reproduce the issue

1.
2.
3.
...

### What version of Home Assistant Core has the issue?

core-2024.3.x

### What was the last working version of Home Assistant Core?

core-2024.2.5

### In which browser are you experiencing the issue with?

Fully Kiosk Browser 1.55

### Which operating system are you using to run this browser?

Android 10

**Comment by ildar170975** [comment]
What kind of cards are you using? Are they custom cards?
Probably you should post a code for the view.

**Comment by pickonedev** [comment]
Does not matter what cards I use, original or custom...
It is because of the stack cards...

**Comment by ildar170975** [comment]
Have you tried purging a frontend cache on devices?

**Comment by pickonedev** [comment]
Yes, I done everything, I have even restarted from factory, the tables

**Comment by karwosts** [comment]
Please use safe mode for your screenshots to reproduce. Your frontend looks heavily themed/custom and we don't triage those issues.

**Comment by karwosts** [comment]
That is not a safe mode screenshot. It doesn't matter that you don't use a custom card, the mere existence of any custom card in resources can break an infinite number of things.

**Comment by pickonedev** [comment]
Alright.

Look, this screenshot it is from computer's browser, cards have a gap/padding between, which is ok, this is how it should look...
![WhatsApp Image 2024-03-26 at 19 11 48_d79127b5](https://github.com/home-assistant/frontend/assets/34201387/5f2ec23e-8bb3-42a9-ad7d-1776432e22ed)


And this is from the huawei tablets with android 10
![WhatsApp Image 2024-03-26 at 19 11 48_82ba826d](https://github.com/home-assistant/frontend/assets/34201387/100a1f66-417a-434e-bff3-cbcee4db2d07)


This is the code

```
type: vertical-stack
cards:
  - show_name: true
    show_icon: true
    type: button
    tap_action:
      action: toggle
    entity: switch.mpower_gaming_headsets_charger
  - type: entity
    entity: >-
      sensor.windows_asus_rog_strix_z790_e_gaming_wifi_ii_nuvoton_nct6798d_voltages_vcore
  - type: light
    entity: light.xiaomi_gateway_light
  - type: markdown
    content: >-
      The **Markdown** card allows you to write any text. You can style it
      **bold**, *italicized*, ~strikethrough~ etc. You can do images, links, and
      more.


      For more information see the [Markdown
      Cheatsheet](https://commonmark.org/help).

```

**Comment by pickonedev** [comment]
If I don't use Stack cards, all the cards have gap/padding between them. Only when I use cards inside Stack cards, the problem appears...
This is why I blame the stack cards

**Comment by ildar170975** [comment]
Can the `gap` property added recently cause this?

Is this property supported in Android 10? 
Here it is specified as supported:
https://caniuse.com/?search=gap
![изображение](https://github.com/home-assistant/frontend/assets/71872483/84a89123-bc3b-45f4-b080-6273ed8b102a)

**Comment by pickonedev** [comment]
I don't know if is supported in my tablet.
I use Fully Kiosk Browser on all my tablets, Andoid 10 and Android 12. But problems I get only with Android 10 tablets...

So... there is something new added, about this issue which I have... If this was the only think added, this could be the problem.
Maybe a condtition which check what device/os/browser you use, and let your device use the proper code, could help. 
It is clear that something from the new HA version, has broken the look on some devices

**Comment by ildar170975** [comment]
> Maybe a condtition which check what device/os/browser you use, and let your device use the proper code, could help.

Would be great. Otherwise we have situations like a recent "my sliders are not shown".

**Comment by pickonedev** [comment]
I have done the test, seems that the tablet does not know about the "gap" css :-(

**Comment by pickonedev** [comment]
This code

```
<div style="background:red;margin:10px">Margin of 10 px</div>
<div style="background:green;margin:20px">Margin of 20 px</div>
<div style="background:red;margin:10px">Margin of 10 px</div>
<div style="background:green;margin:20px">Margin of 20 px</div>
<div style="background:blue;gap:10px">Gap of 10 px</div>
<div style="background:yellow;gap:10px">Gap of 20 px</div>
<div style="background:blue;gap:10px">Gap of 10 px</div>
<div style="background:yellow;gap:10px">Gap of 20 px</div>
```

Result this on my tablet

![WhatsApp Image 2024-03-26 at 19 30 20_0b908edc](https://github.com/home-assistant/frontend/assets/34201387/c654d292-4275-4110-87d4-b355285d3bfd)

**Comment by pickonedev** [comment]
> Can the `gap` property [added recently](https://github.com/home-assistant/frontend/commit/95136993322cb55938a69613bc5ad9918fc3d1da) cause this?
> 
> Is this property supported in Android 10? https://caniuse.com/?search=gap ![изображение](https://private-user-images.githubusercontent.com/71872483/316971841-84a89123-bc3b-45f4-b080-6273ed8b102a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE0NzQzMDQsIm5iZiI6MTcxMTQ3NDAwNCwicGF0aCI6Ii83MTg3MjQ4My8zMTY5NzE4NDEtODRhODkxMjMtYmMzYi00NWY0LWIwODAtNjI3M2VkOGIxMDJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI2VDE3MjY0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY4MTE2YzU2NGNiMDI5YTNhYzhhZjg3ZTg2NzZkZjA5MjdkZGU2MDQ1YzUyZWM1ZGQ5MDc2YzdmYWRmM2YzYmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pnStkGJ8dDf-iv-tNWD47R1ESwFktK8LZmb5_0b6EIA)

Still... I have a question about this...

"Gap" is related to the compatibility of browser, right? But, it is about OS as well?

I ask this because in your photo I only see compatibility of "gap" regarding browsers, not OSs as well... And if is only about browsers, how come that it is working on some newer tablets, with the same browser?

**Comment by pickonedev** [comment]
I'm back with another test I made:

- Fully Kiosk Browser on Android 10 Huawei tablet = Bad display (no padding between cards inside Stack cards)
- Default standard browser of Android 10 Huawei tablet = Good display (there are paddings and everything)

I have tested on Android 11 + Fully Kiosk Browser and everything is good.

Seems that only on Andoird 10 + Fully Kiosk Browser, I meet these problems.

There is any chance that this issue could be fixed in the next update? Or what I should do next?
